### ISSUE TITLE
Unable to play extra contents (New York, New York)

### ISSUE THREAD

**Issue by macgo-dev** [macgo-player/bluray-player#5]
Selecting extra content from the Pop-up Menu is not working.

Title: New York, New York
ASIN: [B004XFVBVU](https://www.amazon.it/dp/B004XFVBVU)
Barcode: 8010312094989

**Comment by macgo-dev** [comment]
This issue should be fixed in the next update, version 3.3.7.

**Comment by macgo-dev** [comment]
However, there is another issue that may cause Pop-up Menu to disappear forever. It seems a race condition because it happens randomly in release build only. I'll fix this issue in further update.

**Comment by macgo-dev** [comment]
The major cause of this issue is a bug related to `HVideoConfigTemplate.SCREEN_RECTANGLE`, which is already fixed in 3.3.7. Another cause is a potential deadlock in overall paint routines.

To avoid flickering caused by incomplete drawing, our implementation of `java.awt.Toolkit.sync()` waits for `org.havi.ui.HScene.paint()` to finish. But there is a `Object.wait()` in `com.thomson.bluray.bdjive.xlet.XletContainer.paint()` blocking the painting. The notifying is expected in another thread, but which is already blocked by `java.awt.Toolkit.sync()`.

Our implementation assumes that `paint()` in xlet is non-blocking, and xlet assumes that `sync()` is non-blocking. To fix the deadlock, I added a timeout of 100ms for waiting `paint()` in `sync()`. The changes will be added in 3.3.8.
